3rd-party/woocommerce.php

```diff
@@ -97,7 +97,17 @@ function jetpack_woocommerce_infinite_scroll_style() {
 function jetpack_woocommerce_lazy_images_compat() {
 	wp_add_inline_script( 'wc-cart-fragments', "
 		jQuery( 'body' ).bind( 'wc_fragments_refreshed', function() {
-			jQuery( 'body' ).trigger( 'jetpack-lazy-images-load' );
+			var jetpackLazyImagesLoadEvent;
+			try {
+				jetpackLazyImagesLoadEvent = new Event( 'jetpack-lazy-images-load', {
+					bubbles: true,
+					cancelable: true
+				} );
+			} catch ( e ) {
+				jetpackLazyImagesLoadEvent = document.createEvent( 'Event' )
+				jetpackLazyImagesLoadEvent.initEvent( 'jetpack-lazy-images-load', true, true );
+			}
+			jQuery( 'body' ).get( 0 ).dispatchEvent( jetpackLazyImagesLoadEvent );
```

Comment:
**Note:** While it may feel like it, switching from jQuery `trigger` to dispatching native events **does not** introduce a regression. There are two notable differences in the usage of both:

1. jQuery `trigger` creates proprietary events that are only observable using jQuery (`on`, `bind` etc.). Meaning that there is no 3rd party code out there using `addEventListener` to observe them.
2. The native `dispatchEvent` is observable using both `addEventListener` and `jQuery` methods. Importantly, jQuery will normalize the incoming event and expose any additional `detail` data in a jQuery way.

This means any pre-existing handler is not going to be affected by a change from the proprietary `trigger` to native `dispatchEvent`. :tada:

Reply:
It seems the [issue reported](https://github.com/Automattic/jetpack/pull/14886#pullrequestreview-377140687) by @jeherve disproves the above. It now seems jQuery listeners (`on`, `bind`) do not pass the `detail` data as a second parameter to event handlers after all. I've however tested that behavior before and I'm not certain why I'm seeing a different outcome now. I will revisit this issue tomorrow.